### ISSUE TITLE
Add cancellation token to YieldUpdatesAsync

### ISSUE THREAD

**Issue by tuscen** [TelegramBots/Telegram.Bot.Extensions.Polling#3]
Current implementation doesn't allow the following common pattern:
```c#
await foreach (var update in receiver.YieldUpdatesAsync().WithCancellation(cancellationToken))
{
    // ... 
}
```

To allow this to work `YieldUpdatesAsync`  needs to have cancellation token as a parameter with an attribute:

```c#
public interface IYieldingUpdateReceiver
{
    int PendingUpdates { get; }
    IAsyncEnumerable<Update> YieldUpdatesAsync([EnumeratorCancellation] CancellationToken cancellationToken = default);
}
```

This will allow uniform cancellation support in `IYieldingUpdateReceiver` implementations instead of ad-hoc params in different implementations.

**Comment by MihaZupan** [comment]
Looks good, I will look into implementing it.

**Comment by tuscen** [comment]
Do you think if there's another way of controlling receiving thread in the queued update receiver other than `StartReceiving`/`StopReceiving`? I've tried to implement cancellation token support to `YieldUpdatesAsync` and with the blocking receiver it was straightforward. But with the queued one the result is a complete mess from an API standpoint when you have two ways of stopping receiving thread.

**Comment by tuscen** [comment]
Fixed in #7
